internal/impl/devbox.go

```diff
@@ -399,59 +399,164 @@ func (d *Devbox) saveCfg() error {
 	return cuecfg.WriteFile(cfgPath, d.cfg)
 }
 
-func (d *Devbox) Services() (plugin.Services, error) {
-	return plugin.GetServices(d.mergedPackages(), d.projectDir)
+func (d *Devbox) Services() (services.Services, error) {
+	svcSet := services.Services{}
+	pluginSvcs, err := plugin.GetServices(d.mergedPackages(), d.projectDir)
+	if err != nil {
+		return svcSet, err
+	}
+
+	userSvcs := services.FromProcessComposeYaml(d.projectDir)
+
+	return lo.Assign(pluginSvcs, userSvcs), nil
+
 }
 
 func (d *Devbox) StartServices(ctx context.Context, serviceNames ...string) error {
 	if !IsDevboxShellEnabled() {
 		return d.RunScript("devbox", append([]string{"services", "start"}, serviceNames...))
 	}
-	return services.Start(ctx, d.mergedPackages(), serviceNames, d.projectDir, d.writer)
+
+	if !services.ProcessManagerIsRunning() {
+		fmt.Fprintln(d.writer, "Process-compose is not running. Starting it now...")
+		fmt.Fprintln(d.writer, "\nNOTE: We recommend using `devbox services up` to start process-compose and your services")
+		return d.StartProcessManager(ctx, serviceNames, true, "")
+	}
+
+	svcSet, err := d.Services()
+	if err != nil {
+		return err
+	}
+
+	if len(svcSet) == 0 {
+		return usererr.New("No services found in your project")
+	}
+
+	for _, s := range serviceNames {
+		if _, ok := svcSet[s]; !ok {
+			return usererr.New(fmt.Sprintf("Service %s not found in your project", s))
+		}
+	}
+
+	for _, s := range serviceNames {
+		err := services.StartServices(ctx, d.writer, s, d.projectDir)
+		if err != nil {
+			fmt.Printf("Error starting service %s: %s", s, err)
+		} else {
+			fmt.Printf("Service %s started successfully", s)
+		}
+	}
+	return nil
+}
+
+func (d *Devbox) StopServices(ctx context.Context, serviceNames ...string) error {
+	if !IsDevboxShellEnabled() {
+		return d.RunScript("devbox", append([]string{"services", "stop"}, serviceNames...))
+	}
+
+	if !services.ProcessManagerIsRunning() {
+		return usererr.New("Process manager is not running. Run `devbox services up` to start it.")
+	}
+
+	if len(serviceNames) == 0 {
+		return services.StopProcessManager(ctx, d.writer)
+	}
+
+	svcSet, err := d.Services()
+	if err != nil {
+		return err
+	}
+
+	for _, s := range serviceNames {
+		if _, ok := svcSet[s]; !ok {
+			return usererr.New(fmt.Sprintf("Service %s not found in your project", s))
+		}
+		err := services.StopServices(ctx, s, d.projectDir, d.writer)
+		if err != nil {
+			fmt.Fprintf(d.writer, "Error stopping service %s: %s", s, err)
+		}
+	}
+	return nil
```

Comment:
do we need to call `services.StopProcessManager` here, since we've stopped all the services that were running?